docs_src/design/adr/core/0022-UoM.md

```diff
@@ -0,0 +1,199 @@
+    values="lbs,ounces,kilos,grams"
+```
+
+!!! Note
+    When the units of measure are defined in-line, then *environmental overrides can be used to specify and override the units of measure in the configuration.toml file*.  However, since the Uom is a map, the unit of measure category must already exist in order to be overridden.  In the example above, you could provide an environmental override to replace `Uom.weights` but you could not provide an override for `Uom.speed` as it does not exist in the original configuration file.
+
+#### Specifying Validation on or off
+Additionally, in either external or inline definition, a configuration option for unit of measure validation being `on` or `off` will be provided (note `validation` in both example A and B above).  The units of measure or the location of the UoM file is static, but the ability to turn validation on/off is dynamic and therefore in the writable area of configuration.  For backward compatibility, validation will be off by default.
+
+Again, when a device resource does not have a `units` specified, this is still considered "valid".  So in reality, the diagram below specifies how validation is done by core metadata.
+
+![image](EdgeX_UoM_Validation.png)
+
+### Validation of the Units of Measure
+
+Core metadata will read the units of measure from its configuration (either inline or from the specified external file).  Like all configuration information, this data will be stored in the configuration service (Consul today) on initial startup of the core metadata service.
+
+When validation is turned `on` (Writable.UoM.validation is set to true), all device profile `units` (in device resource, device properties) will be validated against the list of units of measure by core metadata.  In other words, when a device profile is created or updated or when a device resource is added or updated via the core metadata API, the units specified in the device resource's `units` field (see resource example below) will be checked against the valid list of UoM provided via core metadata configuration.  If the `units` value matches any one of the configuration units of measure, then the device resource is considered valid - allowing the create or update operation to continue.  
+
+If the `units` value does not match any one of the configuration units of measure, then the device profile or device resource operation (create or update) is rejected (error code 500 is returned) and an appropriate error message is returned in the response to the caller of the core metadata API.
+
+!!! Note
+    Importantly (as discussed in Core WG 2/17/22), the `units` field on a profile is and shall remain optional.  If the `units` field is not specified in the device profile, then it is assumed that the device resource does not have well defined units of measure.  In other words, core metadata will not fail a profile with no `units` field specified on a device resource.
```

Comment:
Are we tracking this as something we'd like to change in EdgeX 3.0?

Reply:
Yes - this is already in the EdgeX 3.0 consideration list.  See https://github.com/orgs/edgexfoundry/projects/48